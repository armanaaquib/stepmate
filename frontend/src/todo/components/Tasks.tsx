import React from "react";

import Task from "./Task";
import StyledTasks, { TasksDivider } from "./style/Tasks.style";

const Tasks = (): React.ReactElement => {
  return (
    <StyledTasks>
      <Task />
      <TasksDivider />
      <Task />
    </StyledTasks>
  );
};

export default Tasks;
