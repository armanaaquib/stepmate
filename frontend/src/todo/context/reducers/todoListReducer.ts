import ActionTypes from "../actions/actionsTypes";
import { Action } from "../actions/todoListActions";

type TaskStatus = "TODO" | "IN_PROGRESS" | "DONE";

type Task = {
  id: number;
  text: string;
  status: TaskStatus;
  createdAt: string;
  modifiedAt: string;
};

type Todo = {
  id: number;
  title: string;
  createdAt: string;
  modifiedAt: string;
  userId: number;
  tasks: Task[];
};

type State = { error: boolean; data: Todo[] };

const todoListReducer = (state: State, action: Action): State => {
  switch (action.type) {
    case ActionTypes.LOAD_TODO_LIST: {
      return { ...state, data: action.payload };
    }
  }
};

export { State, Todo };
export default todoListReducer;
