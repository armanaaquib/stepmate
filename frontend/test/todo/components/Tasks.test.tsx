import React from "react";
import { render, screen } from "@testing-library/react";

import Tasks from "../../../src/todo/components/Tasks";
import { TaskType } from "../../../src/todo/context/reducers/todoListReducer";

describe("<Tasks />", () => {
  const entries: TaskType[] = [
    {
      id: 1,
      text: "task text",
      status: "TODO",
      createdAt: "",
      modifiedAt: "",
    },
    {
      id: 2,
      text: "task text 2",
      status: "TODO",
      createdAt: "",
      modifiedAt: "",
    },
  ];

  it("should not render tasks divider if there are no done tasks", () => {
    render(<Tasks entries={entries} />);

    expect(screen.queryByTestId("tasks-divider")).not.toBeInTheDocument();
  });

  it("should render tasks divider if there are done tasks", () => {
    const newEntries: TaskType[] = [
      ...entries,
      { ...entries[0], id: 3, status: "DONE" },
    ];
    render(<Tasks entries={newEntries} />);

    expect(screen.queryByTestId("tasks-divider")).toBeInTheDocument();
  });

  it("should render tasks", () => {
    const newEntries: TaskType[] = [
      ...entries,
      { ...entries[0], id: 3, status: "DONE" },
    ];
    const renderedTasks = render(<Tasks entries={newEntries} />);

    expect(renderedTasks).toMatchSnapshot();
  });

  it("should render no tasks", () => {
    const renderedTasks = render(<Tasks entries={[]} />);

    expect(renderedTasks).toMatchSnapshot();
  });
});
