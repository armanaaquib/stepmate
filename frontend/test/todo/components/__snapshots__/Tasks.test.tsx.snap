// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tasks /> should render no tasks 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
        data-testid="tasks"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsTCUz"
      data-testid="tasks"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Tasks /> should render tasks 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  font-family: "Montserrat",sans-serif;
  font-size: 16px;
  font-weight: 400;
  margin-bottom: 8px;
  background-color: #ffffff;
  color: #000000;
  border-left-width: 2px;
  border-left-style: solid;
  border-left-color: #323232;
  border-radius: 4px;
  padding: 4px;
}

.c2 {
  font-family: "Montserrat",sans-serif;
  font-size: 16px;
  font-weight: 400;
  margin-bottom: 8px;
  background-color: #ffffff;
  color: #323232;
  border-left-width: 2px;
  border-left-style: solid;
  border-left-color: #00BCCD;
  border-radius: 4px;
  padding: 4px;
}

.c1 {
  height: 0.5px;
  background-color: #ffffff;
  border-radius: 5px;
  margin: 12px 0;
}

<body>
    <div>
      <div
        class=""
        data-testid="tasks"
      >
        <div
          class="c0"
        >
          task text
        </div>
        <div
          class="c0"
        >
          task text 2
        </div>
        <div
          class="c1"
          data-testid="tasks-divider"
        />
        <div
          class="c2"
        >
          task text
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsTCUz"
      data-testid="tasks"
    >
      <div
        class="sc-bdfBwQ hUQhyD"
      >
        task text
      </div>
      <div
        class="sc-bdfBwQ hUQhyD"
      >
        task text 2
      </div>
      <div
        class="sc-dlfnbm fZVFmy"
        data-testid="tasks-divider"
      />
      <div
        class="sc-bdfBwQ knBUnh"
      >
        task text
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
